tests/PhpCodeSnifferTestBase.php

```diff
@@ -69,4 +71,38 @@ protected function getFailures(TaskResultInterface $result)
 
         return $failures;
     }
+
+    /**
+     * Tests code to make sure CodeSniffer triggers the appropriate results.
+     *
+     * @param string $file
+     *   Name of the fixture.
+     * @param string $configuration
+     *   The name of the configuration to use in the task
+     * @param int $expectedFailures
+     *   Expected result after the test.
+     *
+     * @dataProvider dataProvider
+     */
+    public function testPhpCodeSnifferDetector(string $file, string $configuration, int $expectedResultCode, array $expectedFailures): void
+    {
+        $collection = new FilesCollection([$this->getFixture($file)]);
+        $context = new RunContext($collection);
+
+        $result = $this->runTask($configuration, 'phpcs', $context);
+        // Check the result code and possible errors.
+        $this->assertEquals($expectedResultCode, $result->getResultCode());
+        $this->assertEquals($expectedFailures, $this->getFailures($result->first()));
+    }
+
+    /**
+     * Provides test cases for testing the PHP_CodeSniffer task.
+     *
+     * @return array
+     *      Test data.
```

Comment:
isn't this too many spaces?

Reply:
Indeed this seems odd, fixed manually. The component uses the PHP library convention to check code. Right now there is no rule for doc commenting identation in PHP library rules. 
Replaced deprecated PSR-2 with PSR-12.